README.adoc

```diff
@@ -75,88 +61,156 @@ It could be omitted as the title already says that it is an experimental feature
-* Use sentence case (you capitalize the first letter, proper nouns, and names, just like you would in a normal sentence) and punctuation.
-* Avoid explaining:
-** the consequences of the returned code.
-** what causes the code to be returned.
-** possible solutions.
+| SeverityLevel
+| One of `INFORMATION`\|`WARNING`
+m| ERROR
 
-⛔Example (contains a possible solution):
-      The provided pattern is unbounded, consider adding an upper limit to the number of node hops.
+| Category
+| One of `DEPRECATION`\|`HINT`\|`UNSUPPORTED`\|`UNRECOGNIZED`\|`SECURITY`\|`GENERIC`\|`PERFORMANCE`\|`TOPOLOGY`
+| N/A
 
-✅Example:
-     The pattern `<pattern>` is unbounded.
+| Description/Message
+2+| Contains the main information, such as the problem, cause, consequences, a simple solution, or a reference to the Status Codes documentation.
 
-⛔Example:
-This feature is deprecated and will be removed in future versions.
+| Example (only in Status Codes documentation)
+2+| Contains one or more example queries to illustrate the possible scenarios when this notification code would be returned.
 
-✅Example:
-(Deprecated) The repeated variable length relationship r is bound more than once.
+| Suggestions for improvement (only in Status Codes documentation)
+2+| Contains a possible solution for the provided example query.
+|===
 
-⛔Example (contains a possible solution):
-      The provided label is not in the database.
+The following are guidelines on how to write each of the notification components.
 
-✅Example:
-     The label `Perso` does not exist.
 
+==== StatusCode syntax
 
+All Neo4j status codes follow the syntax `Neo.[Class].[Subclass].[Name]`.
 
-=== Description
+[options="header",cols="h,1,2,3,3,2",]
+|===
+| StatusCode
+| `Neo`
+| `[Class]`
+| `[Subclass]`
+| `[Name]`
+| Example
+
+| Notification
+| `Neo`
+| `ClientNotification`
+| One of `Statement`, `Procedure`, `Schema`, `Database`, `Security`, or `Request`
+| Should be specific and explains what this code refers to, e.g., `ExhaustiveShortestPath`.
+| `Neo.ClientNotification.Statement.ExhaustiveShortestPath`
+
+| Error
+| `Neo`
+| One of `ClientError`, `TransientError`, `DatabaseError`
+| One of `ChangeDataCapture`, `Cluster`, `Database`, `Fabric` (deprecated), `General`, `Procedure`, `Request`, `Routing`, `Schema`, `Security`, `Statement`, `Transaction`.
+| Should be specific and explains what this code refers to, e.g., `RemoteExecutionFailed`.
+| `Neo.DatabaseError.Statement.RemoteExecutionFailed`
+|===
 
-This is the description of the returned code.
 
-The description should contain the following:
+==== Notification title/Error description
 
-* What caused the notification to be returned
-* Why it might be a problem - what the consequences are
-* If possible, a simple solution, otherwise, a reference to the docs using the sentence:
-_See Status Codes documentation for suggestions for improvement._
+The notification title and the error description contain similar information.
```

Comment:
I wonder if we should mention that the error description is normally not exposed to the user through the error (unless there is a fallback where it is used as error message) but only part of this documentation. What do you think?